mllib/src/main/scala/org/apache/spark/ml/tuning/ValidatorParams.scala

```diff
@@ -183,8 +198,15 @@ private[ml] object ValidatorParams {
           val paramPairs = pMap.map { case pInfo: Map[String, String] =>
             val est = uidToParams(pInfo("parent"))
             val param = est.getParam(pInfo("name"))
-            val value = param.jsonDecode(pInfo("value"))
-            param -> value
+            if (!pInfo.contains("isJson") ||
+               (pInfo.contains("isJson") && pInfo("isJson").toBoolean.booleanValue())) {
```

Comment:
style nit: indent line 202 +1 space

Also, could you please add a comment saying that SPARK-21221 introduced the "isJson" field?